pulser-core/pulser/channels.py

```diff
@@ -163,6 +163,10 @@ def __post_init__(self) -> None:
             if param in optional:
                 prelude = "When defined, "
                 valid = value is None
+            elif value is None:
+                raise TypeError(
+                    f"'{param}' can't be None in a 'Local' channel."
```

Comment:
Yes, you are right, if the user completely ignores the type hints. I made it take the addressing from `self.addressing` instead of assuming it will be `Local`